go/vt/vtgate/planbuilder/testdata/filter_cases.json

```diff
@@ -6696,5 +6696,69 @@
         "user.user"
       ]
     }
+  },
+  {
+    "comment": "must merge subquery with the right side of the join",
+    "query": "select 1 from unsharded join user u1 where exists (select 1 from unsharded u2 where u1.bar = u2.baz)",
+    "v3-plan": "VT12001: unsupported: cross-shard correlated subquery",
+    "gen4-plan": {
+      "QueryType": "SELECT",
+      "Original": "select 1 from unsharded join user u1 where exists (select 1 from unsharded u2 where u1.bar = u2.baz)",
+      "Instructions": {
+        "OperatorType": "SemiJoin",
+        "JoinVars": {
+          "u1_bar": 0
+        },
+        "ProjectedIndexes": "-2",
+        "TableName": "unsharded_`user`_unsharded",
+        "Inputs": [
+          {
+            "OperatorType": "Join",
+            "Variant": "Join",
+            "JoinColumnIndexes": "R:0,L:0",
+            "TableName": "unsharded_`user`",
+            "Inputs": [
```

Comment:
for the input to semi join it receives.

u1.bar, 1

what does -2 in projected indexes represent?

on semi join we will have 
u1.bar, 1, 1 and take the last column `2` where `-` stands for right.

Is my understanding correct?

Reply:
The semi join has the join between `unsharded` and `user` as the input, and it then checks with the last route if each row in the input can be returned or not. The data sent from the LHS to the RHS is `u1.bar`, which we can see here:

```
        "JoinVars": {
          "u1_bar": 0
        },
```

`"ProjectedIndexes": "-2"` on the semi-join means that we will output the second column from the left. Semi-joins never return data from the RHS, so these numbers will always be negative. We can check the second row on the join columns `"JoinColumnIndexes": "R:0,L:0",`, which tells us that the second column is the first column coming from the LHS of the inner join, which is the literal `1`.

Does that make sense?